### PR TITLE
Implement caching

### PR DESCRIPTION
**Pull request by sergeyfrolov**

### CODE REVIEW

File: rust-src/src/flow_tracker.rs [hunk 1]
```diff
@@ -49,8 +51,38 @@ pub struct FlowTracker {
...
+    mcache: MeasurementCache,
+}
+
+// to ease load on db, cache requests
+pub struct MeasurementCache {
+    pub last_flush: time::Tm,
+    measurements: HashMap<(i64, i32), i32>,
 }
 
+const MEASUREMENT_CACHE_FLUSH: i64 = 60; // every min
```
Comment (ewust) [inline rust-src/src/flow_tracker.rs]: You might want a time::Duration here, then you won't have to convert to .sec later.

File: rust-src/src/flow_tracker.rs [hunk 2]
```diff
@@ -162,22 +192,33 @@ ON CONFLICT (id) DO NOTHING;") {
...
+                            &fp.elliptic_curves, &fp.ec_point_fmt, &fp.sig_algs, &fp.alpn]);
+
+                        // once in a while -- flush
+                        if curr_time.to_timespec().sec - self.mcache.last_flush.to_timespec().sec >
+                            MEASUREMENT_CACHE_FLUSH {
+                            let insert_measurement = match db_conn.prepare("INSERT
 INTO measurements (unixtime, id, count)
 VALUES ($1, $2, $3)
 ON CONFLICT ON CONSTRAINT measurements_pkey1 DO UPDATE
   SET count = measurements.count + 1;") {
```
Comment (ewust) [inline rust-src/src/flow_tracker.rs]: ON CONFLICT you'll have to add more than just 1 to the measurements.count, you have to add count. Right now, this only gets the correct count if it's an INSERT, but won't add the correct value on update. Probably want `SET count = measurments.count + $4;`

File: rust-src/src/flow_tracker.rs [hunk 3]
```diff
@@ -162,22 +192,33 @@ ON CONFLICT (id) DO NOTHING;") {
...
-                                println!("Preparing insert_measurement failed: {}", e);
-                                return;
+                                Ok(stmt) => stmt,
+                                Err(e) => {
+                                    println!("Preparing insert_measurement failed: {}", e);
+                                    return;
+                                }
+                            };
+                            for (k, count) in self.mcache.flush() {
+                                insert_measurement.execute(&[&(k.0), &(k.1), &(count)]);
```
Comment (ewust) [inline rust-src/src/flow_tracker.rs]: and then repeat the &(count) parameter:
`insert_measurement.execute(&[&(k.0), &(k.1), &(count), &(count)]);`

File: rust-src/src/flow_tracker.rs (hunk 1 above)
Comment (sergeyfrolov) [reply rust-src/src/flow_tracker.rs]: Tried to do this first, but initialization of time::Duration via function new() is no allowed for const

**Pull request merged**